tests/test_sftp.py

```diff
@@ -769,3 +770,36 @@ def test_write_memoryview(self, sftp):
                 assert f.read() == data
         finally:
             sftp.remove('%s/write_memoryview' % sftp.FOLDER)
+
+
+class SlowChannelServer (StubServer):
+    def check_channel_request(self, kind, chanid):
+        time.sleep(10)
+        return paramiko.OPEN_SUCCEEDED
+
+@slow
+class TestSFTPTimeout(object):
+    def test_auth_session_timeout(self, sftp_server):
+        """
+        verify that the open_sftp has a configurable timeout
+        """
+
+        # copied from conftest.sftp_server
```

Comment:
Was there a reason this was copied instead of just used directly?

Reply:
That function connects the client socket directly to a `Transport` and returns it. I needed just the socket, so I can pass it into `SSHClient` to test `SSHClient.open_sftp` and all the other functions it calls.

Now that I think about it: I could split that function into two, a lower level one that returns the client socket, which is used by `conftest.sftp_server` and by this test. Would that make sense?

Reply:
I think so, yea. Give it a shot!